paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -369,7 +371,7 @@ def sync_crypto_secrets(
             # the secret name here must match the secret name given in the secret volume config,
             # i.e. `kubernetes.client.V1SecretVolumeSource`'s `secret_name` must match below
             secret_name=instance_config.get_crypto_secret_name(),
-            secret_data=secret_data,
+            get_secret_data=(lambda: secret_data),
```

Comment:
we probably want a comment here to make it clear why we're doing this :)

Reply:
Added param docstring, now I am having second thoughts that I need to postpone getting vault keys as well. Since `crypto_keys` is not widely used currently and that requires further changes, we can ship this to reduce Vault service load